webpack.config.js

```diff
@@ -15,20 +15,31 @@ const config = {
   },
   externals: {
     'fs': true
+  },
+  devtool: 'source-map',
+  devServer: {
+    // enable https
+    https: process.env.HTTPS === 'true' || false,
+    // allow connections from LAN
+    host: '0.0.0.0',
+    // allow connections using hostname
+    disableHostCheck: true,
   }
 };
 
 if (process.env.NODE_ENV === 'production') {
-  config.plugins = [
-    new webpack.DefinePlugin({ 'process.env.NODE_ENV': JSON.stringify('production') }),
-    new webpack.optimize.ModuleConcatenationPlugin(),
-    new webpack.optimize.UglifyJsPlugin({
-      compress: { warnings: false, unused: false },
-      output: { comments: false }
-    })
-  ]
+  config.mode = 'production';
```

Comment:
webpack 4.x update:
- "mode" added, need to set "production" or "development"
- No necessary to define process.env.NODE_ENV value
- ModuleConcatenationPlugin was deprecated and is now optimization.concatenateModules. It's set to on by default in production mode.
- uglifyjs-webpack-plugin was enabled by default.